draft-steele-cose-merkle-tree-proofs.md

```diff
@@ -89,51 +89,50 @@ This document describes the three data structures necessary to use merkle proofs
 
 # Terminology
 
-Leaf Bytes:
+Leaf:
 
-: A merkle tree leaf is labelled with the cryptographic hash of a sequence of bytes.
-These bytes may be structured as a combination of Payload and Extra Data.
+: A merkle tree leaf is the cryptographic hash of a sequence of bytes that combines Payload and Extra Data.
 
 Merkle Tree:
 
-: A Merkle tree is a tree where every leaf is labelled with the cryptographic hash of a sequence of
-bytes and every node that is not a leaf is labeled with the cryptographic hash of the labels of its child nodes.
+: A Merkle tree is a tree where every leaf is a cryptographic hash of a sequence of
+bytes and every node that is not a leaf is the cryptographic hash of the its child nodes.
 
-Merkle Tree Root:
+Merkle Root:
 
-: A Merkle tree root is the root node of a tree which represents the cryptographic hash that commits to all leaves in the tree.
+: A Merkle root is the root node of a tree; this cryptographic hash is a committment to the content of the tree.
 
 Merkle Tree Algorithm:
 
-: A Merkle tree algorithm specifies how nodes in the tree must be hashed to compute the root node.
+: A Merkle tree algorithm specifies how to construct the tree and how to compute its root.
 
 Payload and Extra Data:
 
-: A payload is data bound to in a Merkle tree leaf.
-The Merkle tree algorithm determines how a payload together with extra data is bound to a leaf.
-The simplest case is that the payload is the leaf itself without extra data.
+: A payload is application data used to produce a Leaf.
+The Merkle tree algorithm determines how a payload together with extra data is used to produce a leaf.
+The simplest case is that the leaf is the cryptographic hash of the payload without extra data.
 
 Inclusion Path:
 
-: An inclusion path confirms that a value is a leaf of a Merkle tree known only by its root hash (and tree size, possibly).
+: The inclusion path enables a verifier to recompute a root, given a leaf.
 
-Signed Merkle Tree Proof:
+Signed Inclusion Proof:
 
-: A signed Merkle tree proof is the combination of signed Merkle tree root hash, inclusion path, extra data, and payload.
+: A signed inclusion proof is a combination of payload, extra data, inclusion path and signed envelope that includes a merkle root.
 
 # CBOR Merkle Structures
 
-This section describes representations of merkle tree structures in CBOR.
+This section describes representations of merkle proof structures in CBOR.
 
 Some of the structures such as the construction of a merkle tree leaf,
 or an inclusion proof from a leaf to a merkle root, might have several different representations.
 
 Some differences in representations are necessary to support efficient
-verification of proofs and compatibility with deployed tree algorithms used in specific implementations.
+verification of different kinds of inclusion proofs and for compatibility with deployed tree algorithms used in specific implementations.
 
-## Signed Merkle Tree Root
+## Signed Inclusion Proof
```

Comment:
Can you outline what else you'll change in this PR? Further below there's "Signed Merkle Tree Proof" which seems to be what "Signed Inclusion Proof" will be?

Reply:
Yes, its work in progress. 

Main goal was to take a pass on terminology and address potential confusion or collisions with other concepts.

For example "envelope payload" vs "merkle leaf payload".

We also discussed adding some of the more specialized trees.